components/kubearchive/base/migration-job.yaml

```
@@ -0,0 +1,38 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: kubearchive-schema-migration-
+  namespace: kubearchive
+  annotations:
+    argocd.argoproj.io/hook: PreSync
+spec:
+  parallelism: 1
+  backoffLimit: 4
+  template:
+    spec:
+      containers:
+        - name: migration
+          image: quay.io/fedora/python-311:20240911
+          command: ["/bin/sh", "-c"]
+          args:
+            - git clone https://github.com/kubearchive/kubearchive --depth=1 --branch=${KUBEARCHIVE_VERSION} /tmp/kubearchive;
+              cd /tmp/kubearchive;
+              curl -L https://github.com/golang-migrate/migrate/releases/latest/download/migrate.linux-amd64.tar.gz | tar xvz migrate;
+              ./migrate -verbose -path integrations/database/postgresql/migrations/ -database postgresql://${DATABASE_USER}:${DATABASE_PASSWORD}@${DATABASE_URL}:${DATABASE_PORT}/${DATABASE_DB} up
+          resources:
+            limits:
+              cpu: 10m
+              memory: 64Mi
+            requests:
+              cpu: 10m
+              memory: 64Mi
+          envFrom:
+            # Provide DB URL, user, password, database and port as env vars
+            - secretRef:
+                name: kubearchive-database-credentials
+          env:
+            - name: KUBEARCHIVE_VERSION
+              value: v1.0.1
```

I'd propose to use an environment variable for the version

```suggestion
              curl -L https://github.com/golang-migrate/migrate/releases/${GOLANG_MIGRATE_VERSION}/download/migrate.linux-amd64.tar.gz | tar xvz migrate;
              ./migrate -verbose -path integrations/database/postgresql/migrations/ -database postgresql://${DATABASE_USER}:${DATABASE_PASSWORD}@${DATABASE_URL}:${DATABASE_PORT}/${DATABASE_DB} up
          resources:
            limits:
              cpu: 10m
              memory: 64Mi
            requests:
              cpu: 10m
              memory: 64Mi
          envFrom:
            # Provide DB URL, user, password, database and port as env vars
            - secretRef:
                name: kubearchive-database-credentials
          env:
            - name: KUBEARCHIVE_VERSION
              value: v1.0.1
            - name: GOLANG_MIGRATE_VERSION
              value: latest
```